BuyPolish/Pola/UI/ProductSearch/ScanCode/BPScanCodeViewController.m

```diff
@@ -168,6 +182,46 @@ - (void)setAddingCardEnabled:(BOOL)addingCardEnabled {
     [UIApplication sharedApplication].idleTimerDisabled = _addingCardEnabled;
 }
 
+#pragma mark - Keyboard Controller
+
+- (void) hideKeyboardController {
+    [self.castView.keyboardButton setSelected:NO];
+
+    [self.keyboardViewController willMoveToParentViewController:nil];
+
+    [UIView animateWithDuration:kAnimationTime animations:^{
+        self.castView.infoTextLabel.alpha = self.castView.stackView.cardCount > 0 ? 0.0 : 1.0;
+        self.castView.rectangleView.alpha = 1.0;
+        self.castView.stackView.alpha = 1.0;
+        self.keyboardViewController.view.alpha = 0.0;
+    } completion:^(BOOL finished) {
+        [self.keyboardViewController.view removeFromSuperview];
+        [self.keyboardViewController removeFromParentViewController];
+
+        [self.castView configureInfoLabelForMode:BPScanCodeViewLabelModeScan];
+    }];
```

Comment:
I think that all view related stuff, should be made in `BPScanCodeView`, e.g.:
```
self.keyboardViewController willMoveToParentViewController:nil];

[self.castView hideKeyboardViewWithCompletion:^() {
       [self.keyboardViewController removeFromParentViewController]
}];
```

in `[BPScanCodeView hideKeyboardViewWithCompletion:]` there should be:
```
[self.keyboardButton setSelected:NO];

[UIView animateWithDuration:kAnimationTime animations:^{
        self.infoTextLabel.alpha = self.stackView.cardCount > 0 ? 0.0 : 1.0;
        self.rectangleView.alpha = 1.0;
        self.stackView.alpha = 1.0;
        self.keyboardView.view.alpha = 0.0;
} completion:^(BOOL finished) {
        [self.keyboardView removeFromSuperview];
        self.keyboardView = nil;
        [self configureInfoLabelForMode:BPScanCodeViewLabelModeScan]; 
}];
```